src/Mod/Part/Gui/Command.cpp

```diff
@@ -1309,6 +1309,57 @@ bool CmdPartBoolean::isActive(void)
     return (hasActiveDocument() && !Gui::Control().activeDialog());
 }
 
+/**
+ * Used for Loft, Sweep, Revolve, and Extrude where the user has preselected
+ * some objects prior to executing the command. In some such cases the dialog
+ * can be skipped, for example, selecting a sketch in the tree, a path
+ * in the 3d view, and then clicking the Sweep toolbar icon
+ */
+
+namespace PartGui {
+
+bool callFunction (const Py::Object& module, const std::string& function, App::Document* doc, const char* text) {
+    try {
+        doc -> openTransaction(text);
+        Py::Boolean result(module.callMemberFunction(function));
+        bool ok = result.as_bool();
+        ok ? doc->commitTransaction() : doc->abortTransaction();
+        return ok;
+    }
+    catch (const Py::Exception&) {
+        doc->abortTransaction();
+        throw; //re-throw the exception
+    }
+}
+
+void checkSkipDialog (Gui::TaskView::TaskDialog *dlg, const std::string& function, const char* text) {
+
+    bool neverSkipDialog = App::GetApplication().GetUserParameter().GetGroup("BaseApp/Preferences/Mod/Part")->GetBool("NeverSkipDialog", false);
+    if (!neverSkipDialog) {
+        App::Document* doc = App::GetApplication().getActiveDocument();
+        Base::PyGILStateLocker lock;
+        try {
+            PyObject* module = PyImport_ImportModule("BasicShapes.SkipGui");
+            if (!module) {
+                throw Py::Exception();
+            }
+            Py::Module skipgui(module, true);
+            if (!PartGui::callFunction(skipgui, function, doc, text)){
+                Gui::Control().showDialog(dlg); //show the dialog on selection error
+            }
+        }
+        catch (Py::Exception&) {
+            Base::PyException e;
+            e.ReportException();
+            Gui::Control().showDialog(dlg); //show the dialog on runtime python error
+        }
+    } else {
+        Gui::Control().showDialog(dlg); //user parameter set to never skip the dialog
+    }

```

Comment:
Maybe you could return if `callFunction` succeeds, then `showDialog` only once as a default behavior at the end of he function.

Reply:
Done.